Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,912 @@
+
+Foo {
+public function /* test */ __construct($param) {
+}
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+class  /* test */
+// another
+
+Foo {
+public function /* test */ Foo($param) {
+}
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testAlphaBeta()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        echo 'alpha';
+    }
+    public function __construct()
+    {
+        echo 'beta';
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testAlphaBetaTrick1()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        // This is not $this->__construct()
+        echo 'alpha';
+    }
+    public function __construct()
+    {
+        echo 'beta';
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testAlphaBetaTrick2()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        echo 'alpha';
+    }
+    public function __construct()
+    {
+        // This is not $this->Foo()
+        echo 'beta';
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testAlphaBetaTrick3()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        echo 'alpha';
+        /* yeah, ok let's construct it anyway */
+        $this->__construct();
+    }
+    public function __construct()
+    {
+        echo 'beta';
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testAlphaBetaTrick4WithAnotherClass()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        echo 'alpha';
+    }
+    public function __construct()
+    {
+        $this->Foo();
+        // Do something more!
+        echo 'beta';
+    }
+}
+
+Class Bar
+{
+    function __construct()
+    {
+        $this->foo = 1;
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    public function Foo()
+    {
+        echo 'alpha';
+    }
+    public function __construct()
+    {
+        $this->Foo();
+        // Do something more!
+        echo 'beta';
+    }
+}
+
+Class Bar
+{
+    function bar()
+    {
+        $this->foo = 1;
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testAbstract()
+    {
+        $expected = <<<'EOF'
+<?php
+
+abstract class Foo
+{
+    abstract function Foo();
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testAbstractTrick()
+    {
+        $expected = <<<'EOF'
+<?php
+
+abstract class Foo
+{
+    abstract function Foo();
+
+    public function bar()
+    {
+        // This is messed up, I know
+        $this->__construct();
+    }
+
+    public function __construct()
+    {
+        $this->baz = 1;
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testRevive1()
```

Comment:
Some tests previously contained code found in the field (Revive Adserver). I guess I've made them generic, but left a reference in the test name.